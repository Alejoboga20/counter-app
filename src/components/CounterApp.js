import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ title, subtitle, value }) => {
  const [count, setCount] = useState(0);

  const handleAdd = () => {
    setCount(count + 1);
  };

  const handleSubstrac = () => {
    setCount(count - 1);
  };

  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>Count: {count}</p>
      <div>
        <button onClick={handleAdd}>+</button>
        <button onClick={handleSubstrac}>-</button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired
};

CounterApp.defaultProps = {
  subtitle: 'First App using React JS'
};
export default CounterApp;
