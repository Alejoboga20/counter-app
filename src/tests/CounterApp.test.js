import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../../src/components/CounterApp';

describe('Counter App Tests', () => {
  test('should display CounterApp properly', () => {
    const wrapper = shallow(<CounterApp title='React Counter App' value={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show subtitle from props', () => {
    const subtitle = 'Subtitle from props';
    const wrapper = shallow(
      <CounterApp title='React Counter App' value={0} subtitle={subtitle} />
    );
    const text = wrapper.find('h2').text();
    expect(text).toBe(subtitle);
  });

  test('should show initial count value from props', () => {
    const initialValue = 10;
    const wrapper = shallow(
      <CounterApp title='React Counter App' value={initialValue} />
    );
    const text = wrapper.find('p').text();
    expect(text).toBe(`Count: ${initialValue}`);
  });
});
