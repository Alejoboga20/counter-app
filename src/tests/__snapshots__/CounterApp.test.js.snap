// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter App Tests should display CounterApp properly 1`] = `
<Fragment>
  <h1>
    React Counter App
  </h1>
  <h2>
    First App using React JS
  </h2>
  <p>
    Count: 
    0
  </p>
  <div>
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
    <button
      onClick={[Function]}
    >
      Reset Count
    </button>
  </div>
</Fragment>
`;
